test/distributed/checkpoint/test_tp_checkpoint.py

```diff
@@ -66,16 +79,57 @@ def test_tp_checkpoint(self):
         for param1, param2 in zip(original_state_dict.values(), state_dict.values()):
             self.assertNotEqual(param1.to_local(), param2.to_local())
 
-        dist_cp.load_state_dict(
+        DCP.load_state_dict(
             state_dict=state_dict,
-            storage_reader=dist_cp.FileSystemReader(CHECKPOINT_DIR),
+            storage_reader=DCP.FileSystemReader(CHECKPOINT_DIR),
             planner=DefaultLoadPlanner(),
         )
 
         # now load from checkpoint to check current model parameters are the same as original_state_dict
         for param1, param2 in zip(original_state_dict.values(), state_dict.values()):
             self.assertEqual(param1.to_local(), param2.to_local())
 
+    @with_comms
+    @skip_if_lt_x_gpu(2)
+    @with_temp_dir
+    def test_tp_checkpoint_load_on_meta_device(self):
+        CHECKPOINT_DIR = self.temp_dir
+        mesh_shpe = (self.world_size,)
+        tp_mesh = init_device_mesh(self.device_type, mesh_shpe)
+
+        # create model and move it to GPU with id rank
+        model = UnevenShardedModel(self.device_type).cuda(self.rank)
+        # Parallelize the module based on the given Parallel Style.
+        parallelize_plan = {
+            "net1": RowwiseParallel(),
+            "net2": ColwiseParallel(),
+        }
+        model = parallelize_module(model, tp_mesh, parallelize_plan=parallelize_plan)
+        original_state_dict = deepcopy(model.state_dict())
+
+        DCP.save_state_dict(
```

Comment:
Might be a good opportunity to use the new Checkpointer class

Reply:
Feel free to disregard since we are making Checkpointer private

Reply:
Ye. This was created back in November so I was still using our legacy setup. Can update it later.